exporters/trace_stackdriver/src/main/java/io/opencensus/trace/exporter/StackdriverV1ExporterHandler.java

```diff
@@ -0,0 +1,164 @@
+    traceServiceBlockingStub =
+        TraceServiceGrpc.newBlockingStub(channel)
+            .withCallCredentials(MoreCallCredentials.from(credentials));
+  }
+
+  Trace generateTrace(SpanData spanData) {
+    SpanContext context = spanData.getContext();
+    TraceSpan.Builder spanBuilder =
+        TraceSpan.newBuilder()
+            .setSpanId(toSpanIdProto(context.getSpanId()))
+            .setKind(toSpanKindProto(spanData))
+            .setName(spanData.getName())
+            .setStartTime(toTimestampProto(spanData.getStartTimestamp()))
+            .setEndTime(toTimestampProto(spanData.getEndTimestamp()));
+
+    if (spanData.getParentSpanId() != null && spanData.getParentSpanId().isValid()) {
+      spanBuilder.setParentSpanId(toSpanIdProto(spanData.getParentSpanId()));
+    }
+
+    for (Map.Entry<String, AttributeValue> label :
+        spanData.getAttributes().getAttributeMap().entrySet()) {
+      spanBuilder.putLabels(label.getKey(), attributeValueToString(label.getValue()));
+    }
+
+    // Add Status as labels in the v1 API.
+    spanBuilder.putLabels(STATUS_CODE, spanData.getStatus().getCanonicalCode().toString());
+    if (spanData.getStatus().getDescription() != null) {
+      spanBuilder.putLabels(STATUS_DESCRIPTION, spanData.getStatus().getDescription());
+    }
+
+    Trace.Builder traceBuilder =
+        Trace.newBuilder()
+            .setProjectId(projectId)
+            .setTraceId(toTraceIdProto(context.getTraceId()))
+            .addSpans(spanBuilder.build());
+
+    return traceBuilder.build();
+  }
+
+  private static long toSpanIdProto(SpanId spanId) {
+    return ByteBuffer.wrap(spanId.getBytes()).getLong();
+  }
+
+  private static String toTraceIdProto(TraceId traceId) {
+    return BaseEncoding.base16().lowerCase().encode(traceId.getBytes());
+  }
+
+  private static TraceSpan.SpanKind toSpanKindProto(SpanData spanData) {
+    if (Boolean.TRUE.equals(spanData.getHasRemoteParent())) {
+      return TraceSpan.SpanKind.RPC_SERVER;
+    }
+
+    // This is a hack because the v2 API does not have SpanKind. When switch to v2 this will be
+    // fixed.
+    if (spanData.getName().startsWith("Sent.")) {
+      return TraceSpan.SpanKind.RPC_CLIENT;
+    }
+
+    return TraceSpan.SpanKind.SPAN_KIND_UNSPECIFIED;
+  }
+
+  private static com.google.protobuf.Timestamp toTimestampProto(Timestamp timestamp) {
+    return com.google.protobuf.Timestamp.newBuilder()
+        .setSeconds(timestamp.getSeconds())
+        .setNanos(timestamp.getNanos())
+        .build();
+  }
+
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    return attributeValue.match(
+        new Function<String, String>() {
+          @Override
+          public String apply(String stringValue) {
+            return stringValue;
+          }
+        },
+        new Function<Boolean, String>() {
+          @Override
+          public String apply(Boolean booleanValue) {
+            return booleanValue.toString();
```

Comment:
Suggestion: You could use `com.google.common.base.Functions.toStringFunction()`.

Reply:
That is java8, we are old school 🥇

Reply:
Sorry, I forgot that we use a different Function class.